src/main/java/org/mule/module/apikit/validation/body/form/transformation/MultipartFormDataTextParameter.java

```diff
@@ -23,8 +25,13 @@ public MultipartFormDataTextParameter(String body) {
 
   @Override
   public void validate(Parameter expected) throws InvalidFormParameterException {
-    if (!expected.validate(body)) {
+    if (expected.isArray()) {
+      if (!expected.validateArray(Collections.singletonList(body))) {
+        throw new InvalidFormParameterException(expected.message(body));
+      }
+    } else if (!expected.validate(body)) {
```

Comment:
@dcourtil Asking because I don't remember. Why can't we have this logic on the `Parameter`?

I mean something like
```java
class Whatever implements Parameter {
  bool validate(String value) {
    return isArray() ? validateArray(value) : validateSingle(value);
  }
  // ...
}
```